### PR TITLE
Form controls - `Toggle` component (07)

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

> **Note**
> This is not the final version: we will complete the component (documentation included) when the design specs are finalized.

This PR adds the `Toggle` form controls to the system in two variants:
- base
- field

### :hammer_and_wrench: Detailed description

In this PR @didoo and I have:
- added the "base" and "field" `Toggle` form controls as components
- added the associated documentation page
- added integration and visual regression testing

[Preview of the documentation page](https://hds-components-git-form-controls-07-toggle-hashicorp.vercel.app/components/form/toggle)

Other notes:
 - the page is not linked in the index file of the dummy/scrappy website on purpose, so it's not visible to the public.
 - the design of this component is not yet finalised, but we raised this PR early and we expect to adjust the styling accordingly

### :link: External links

***

### 👀 How to review

👉 Review commit-by-commit

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by didoo — APPROVED** [review]

**Review by MelSumner — APPROVED** [review]
[issue, non-blocker] I think that we should consider asserting if no value has been provided, since it's both required and consistent with how we've handled other things of this nature so far. However, I've already commented on a separate PR about this issue so it's likely to be a change across several components if we decide to do this, so I won't hold up this PR for that reason.

**Review by didoo — COMMENTED** [review]
@alex-ju I've pushed a small change. for me is good to go, when you want feel free to merge.

**Pull request merged**
